torch/__init__.py

```diff
@@ -62,22 +62,27 @@
     else:
         cuda_path = ''
 
-    if sys.version_info >= (3, 8):
-        dll_paths = list(filter(os.path.exists, [th_dll_path, py_dll_path, nvtoolsext_dll_path, cuda_path]))
+    kernel32 = ctypes.WinDLL('kernel32.dll', use_last_error=True)
+    dll_paths = list(filter(os.path.exists, [th_dll_path, py_dll_path, nvtoolsext_dll_path, cuda_path]))
 
-        for dll_path in dll_paths:
+    for dll_path in dll_paths:
+        if sys.version_info >= (3, 8):
             os.add_dll_directory(dll_path)
-
-    if is_conda or sys.version_info < (3, 8):
-        dll_paths = [th_dll_path, py_dll_path, nvtoolsext_dll_path, cuda_path]
-        dll_paths = list(filter(os.path.exists, dll_paths)) + [os.environ['PATH']]
-
-        os.environ['PATH'] = ';'.join(dll_paths)
+        else:
```

Comment:
To be super careful here and support older versions of windows, why not check that kernel32 actually has the method
```suggestion
        elif hasattr(kernel32, 'AddDllDirectory'):
```

Reply:
This one sounds reasonable.